docs/blog-post.md

```diff
@@ -0,0 +1,286 @@
+- name: org.random.project.Data.ClassTest
+  typeSymbol: class
+  fields:
+  - name: a
+    typeName: java.lang.String
+  - name: b
+    typeName: scala.Int
+  - name: c
+    typeName: scala.Double
+  parents:
+  - org.random.project.Data
+- name: org.random.project.Data.ClassWithAdditionData
+  typeSymbol: class
+  fields:
+  - name: ad
+    typeName: org.random.project.Data.AdditionalData
+  parents:
+  - org.random.project.Data
+```
+
+Then, simple `diff` command can be used to check the difference between the version of a schema from `develop`/`main` branch and the
+version from the current commit.
+
+<img width="1043" alt="easy-to-diff" src="https://user-images.githubusercontent.com/49597713/172350313-cd03f033-ca68-40c3-a4df-064456c483d0.png">
+
+### 3. Jackson Akka Serializer drawbacks
+
+Using Jackson Serializer for akka-persistence is also one of the pitfalls. Akka Serialization Helper provides an alternative - a more Scala-friendly serializer that uses [Circe](https://circe.github.io/circe/).
+
+Examples below show some problems that might occur when combining Jackson with Scala code:
+
+#### Example 1 
+Dangerous code for Jackson:
+
+```scala
+case class Message(animal: Animal) extends MySer
+
+sealed trait Animal
+
+final case class Lion(name: String) extends Animal
+final case class Tiger(name: String) extends Animal
+```
+This seems to be all right, but unfortunately will not work with Jackson serialization - in runtime there will be an exception with message like: "Cannot construct instance of `Animal`(...)" - as abstract types need to be mapped to concrete types explicitly in code. So, to make this code work, a lot of Jackson annotations should be added:
+
+```scala
+case class Message(animal: Animal) extends MultiDocPrintService
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonSubTypes(
+  Array(
+    new JsonSubTypes.Type(value = classOf[Lion], name = "lion"),
+    new JsonSubTypes.Type(value = classOf[Tiger], name = "tiger")))
+sealed trait Animal
+
+final case class Lion(name: String) extends Animal
+final case class Tiger(name: String) extends Animal
+
+```
+
+#### Example 2
+
+If a Scala object is defined:
+```scala
+case object Tick
+```
+
+There **will not be exceptions** during serialization but Jackson will create
+another instance of `Tick` instead of restoring the `object Tick`'s underlying singleton. So, deserialization will end up very bad...
+
+```scala
+actorRef ! Tick
+
+// Inside the actor:
+def receive = {
+  case Tick => // this won't get matched !!
+} // message will be unhandled !!
+```
+
+#### But there is a better way to do serialization!
+To get rid of such problems, our circe-based Akka serializer can be used. [Circe Akka Serializer](https://github.com/VirtusLab/akka-serialization-helper/tree/main/circe-akka-serializer) (part of the `akka-serialization-helper` toolbox) uses Circe codecs, derived using [Shapeless](https://circe.github.io/circe/codecs/auto-derivation.html),
```

Comment:
Same as with _Akka_, pls use upper case for Circe consistently.

Reply:
Done